### PR TITLE
fixed live diagnostic argId leak

### PR DESCRIPTION
**Pull request by heejaechang**
it looks like my previous update 2 leak fix for Task introduced argId leaks. since this was much smaller leak than before I didn't notice it being leaked.

fix https://github.com/dotnet/roslyn/issues/10655

### CODE REVIEW

**Comment by heejaechang** [comment]
@dotnet/roslyn-analysis  @mavasani @srivatsn can you take a look?

File: src/VisualStudio/Core/Def/Implementation/TableDataSource/OpenDocumentTracker.cs [hunk 1]
```diff
@@ -43,28 +45,78 @@ public void TrackOpenDocument(DocumentId documentId, object id, AbstractTableEnt
...
-                    AbstractTableEntriesSnapshot<T> snapshot;
-                    if (!weakSnapshot.TryGetTarget(out snapshot))
+                    if (solution.GetDocument(documentId) != null)
                     {
+                        // document still exist.
                         continue;
                     }
 
-                    snapshot.StopTracking();
+                    StopTracking_NoLock(documentId);
```
Comment (mavasani) [inline src/VisualStudio/Core/Def/Implementation/TableDataSource/OpenDocumentTracker.cs]: Do we need to stop tracking the removed project ID(s) as well?

Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/TableDataSource/OpenDocumentTracker.cs]: this track only documents.

**Comment by heejaechang** [comment]
hotfix PR - https://github.com/dotnet/roslyn/pull/10731

**Comment by heejaechang** [comment]
@mavasani so are you okay with the change?

**Comment by mavasani** [comment]
👍

**Pull request merged**